dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/readerOMC.py

```diff
@@ -718,88 +718,73 @@ def find_variable_from_name(self, name):
-                if start != '':
-                    var.set_start_text( [start] )
-                    var.set_use_start(list_string[0].getAttribute('useStart'))
-                else:
-                    var.set_use_start("false")
+        xml_parser = lxml.etree.XMLParser(remove_comments=True)
+        xml_tree = lxml.etree.parse(self.init_xml_file, xml_parser)
+
+        root_element = xml_tree.getroot()
+        if root_element.tag != "fmiModelDescription":
+            raise FmiModelDescriptionElementNotFound(root_element.tag)
+
+        for fmi_model_description_element in root_element:
+            if fmi_model_description_element.tag == "ModelVariables":
+                self.parse_scalar_variables(fmi_model_description_element)
+            elif fmi_model_description_element.tag == "DefaultExperiment":
+                pass # nothing to do
+            else:
+                raise UnknownFmiModelDescriptionElement(fmi_model_description_element.tag)
 
-            # Vars after the read of *_init.xml
-            self.list_vars.append(var)
-            self.dic_vars[var.get_name()] = len(self.list_vars) - 1
+    ##
+    # Parse every ScalarVariable elements contained in ModelVariables element
+    # @param self : object pointer
+    # @param model_variables_elements : ModelVariables etree element
+    # @return
+    def parse_scalar_variables(self, model_variables_elements):
+        for scalar_variable_element in model_variables_elements:
+            if scalar_variable_element.tag == "ScalarVariable":
+                var = Variable()
+                var.set_name(scalar_variable_element.attrib['name'])
+                var.set_variability(scalar_variable_element.attrib['variability'])  # "continuous", "discrete"
+                var.set_causality(scalar_variable_element.attrib['causality'])  # "output" or "internal"
+                var.set_type(scalar_variable_element.attrib['classType'])
+
+                # Vars represented by an output var have an attribute
+                # 'aliasVariable', which is the name of the output var
+                if scalar_variable_element.attrib['alias'] == "alias":
+                    var.set_alias_name(scalar_variable_element.attrib['aliasVariable'], False)
+                if scalar_variable_element.attrib['alias'] == "negatedAlias":
+                    var.set_alias_name(scalar_variable_element.attrib['aliasVariable'], True)
+
+                # 'Start' attribute: this attribute is only assigned for real vars
+                if var.get_type()[0] == "r":
+                    if len(scalar_variable_element) != 1:
+                        print("pb : read_init_xml real on variable " + var.get_name())
+                        sys.exit()
+                    real_scalar_variable_element = scalar_variable_element[0]
+                    if real_scalar_variable_element.tag == 'Real' or real_scalar_variable_element.tag == 'Boolean':
```

Comment:
or real_scalar_variable_element.tag == 'Boolean'

may be useless but this logic was in the previous code